lib/simplefootball_web/scraper/tm_matchday_scraper.ex

```diff
@@ -0,0 +1,153 @@
+defmodule SimplefootballWeb.TMMatchdayScraper do
+  import HTTPoison
+  import Meeseeks.XPath
+  require Logger
+  use Timex
+
+  alias SimplefootballWeb.{
+    Match,
+    Team,
+    MatchRepo,
+    TeamRepo,
+    StringHelpers,
+    ArrayHelpers,
+    MatchdayRepo
+  }
+
+  def scrape_matchday(data, season_id, number) do
+    document = Meeseeks.parse(data)
+
+    {:ok, matchday} =
+      MatchdayRepo.update_or_create_matchday_by(
```

Comment:
Ich kann mir vorstellen, dass es auf Dauer gut sein könnte, wenn mehr "separation of concerns" Einzug hält.

In der aktuellen Implementierung hat TMMatchdayScraper mindestens drei Aufgaben und ist damit abhängig von all diesem Code:

* TM scrapen
* RM resourcen in SimpleFootball resourcen mappen
* SimpleFootball datenbank befüllen

Spontan würde ich mal das Anti Corruption Layer pattern empfehlen.

Hier stand auch kurz ein länglicher Text, der beschreibt wie man es umsetzen könnte. Aber das war direkt totaler Overkill ^^